pkg/connection/store/kubernetes/store.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"context"
+	"encoding/json"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/crossplane/crossplane-runtime/pkg/connection/store"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// Error strings.
+const (
+	errGetSecret            = "cannot get secret"
+	errDeleteSecret         = "cannot delete secret"
+	errUpdateSecret         = "cannot update secret"
+	errCreateOrUpdateSecret = "cannot create or update connection applicator"
+	errParseMetadata        = "cannot parse metadata"
+
+	errExtractKubernetesAuthCreds = "cannot extract kubernetes auth credentials"
+)
+
+type secretMetadata struct {
+	Labels      map[string]string `json:"labels"`
+	Annotations map[string]string `json:"annotations"`
+	Type        corev1.SecretType `json:"type"`
+}
+
+// SecretStore is a Kubernetes Secret Store.
+type SecretStore struct {
+	client     client.Client
+	applicator resource.Applicator
+
+	defaultNamespace string
+}
+
+// NewSecretStore returns a new Kubernetes SecretStore.
+func NewSecretStore(ctx context.Context, local client.Client, cfg v1.SecretStoreConfig) (*SecretStore, error) {
+	if cfg.Kubernetes == nil {
+		// No KubernetesSecretStoreConfig provided, local API Server will be
+		// used as Secret Store.
+		return &SecretStore{
+			client: local,
+			applicator: resource.NewApplicatorWithRetry(resource.NewAPIPatchingApplicator(local),
+				resource.IsAPIErrorWrapped, nil),
```

Comment:
Nit: Feel free to ignore since this is super subjective and doesn't have strong guidance from Go, but I'd personally prefer to avoid line breaks like these in the middle of a function call. I'd either leave the line long or wrap like:

https://github.com/golang/go/wiki/CodeReviewComments#line-length